File: packages/dotnet/src/types/TavoTypes.cs
using System;
using System.Collections.Generic;

namespace TavoAI
{
    /// <summary>
    /// Pagination information
    /// </summary>
    public class PaginationInfo
    {
        public int Page { get; set; }
        public int Limit { get; set; }
        public int Total { get; set; }
        public int Pages { get; set; }
        public bool HasNext { get; set; }
        public bool HasPrev { get; set; }
    }

    /// <summary>
    /// Paginated response wrapper
    /// </summary>
    /// <typeparam name="T">Type of items in the response</typeparam>
    public class PaginatedResponse<T>
    {
        public List<T> Data { get; set; } = new List<T>();
        public PaginationInfo Pagination { get; set; } = new PaginationInfo();
    }

    /// <summary>
    /// List response wrapper
    /// </summary>
    /// <typeparam name="T">Type of items in the response</typeparam>
    public class ListResponse<T>
    {
        public List<T> Data { get; set; } = new List<T>();
        public int Count { get; set; }
    }

    /// <summary>
    /// Error response
    /// </summary>
    public class ErrorResponse
    {
        public string Error { get; set; } = "";
        public string Message { get; set; } = "";
        public object? Details { get; set; }
    }
}

File: packages/java/src/types/TavoTypes.java
package net.tavoai.sdk;

import java.util.List;
import java.util.Map;

/**
 * Common types for Tavo SDK
 */
public class TavoTypes {

    /**
     * Pagination information
     */
    public static class PaginationInfo {
        public int page;
        public int limit;
        public int total;
        public int pages;
        public boolean hasNext;
        public boolean hasPrev;
    }

    /**
     * Paginated response wrapper
     */
    public static class PaginatedResponse<T> {
        public List<T> data;
        public PaginationInfo pagination;
    }

    /**
     * List response wrapper
     */
    public static class ListResponse<T> {
        public List<T> data;
        public int count;
    }

    /**
     * Error response
     */
    public static class ErrorResponse {
        public String error;
        public String message;
        public Object details;
    }
}
